Android: Changed FAB Placement and Video Settings Icon

sonic2kk opened the pull request:
- The Floating Action Button was set to be 32dp instead of the common 16dp.
- Added padding in the MainActivity Toolbar to make it look (in my opinion) cleaner.
- Changed Floating Action Button style from "auto" to "normal".
- Changed buildToolsVersion in app build.gradle from 25.0.2 to 25.0.1, as this was causing an error stating that AIDL was missing
- Uncommented some cmake build.gradle lines to allow the project to be ran on a physical device
- Changed Video Settings icon to something more obvious (https://material.io/icons/#ic_tv)
- Changed theme from DolphinGamecube to DolphinBase

Before and after: http://imgur.com/a/Oeukv

Helios747 commented on Source/Android/app/build.gradle:
```diff
@@ -64,8 +64,8 @@ android {
             cmake {
-                arguments "-DANDROID_STL=c++_static", "-DCMAKE_BUILD_TYPE=RelWithDebInfo" // , "-DENABLE_GENERIC=ON"
-                abiFilters "arm64-v8a" //, "armeabi-v7a", "x86_64", "x86"
+                arguments "-DANDROID_STL=c++_static", "-DCMAKE_BUILD_TYPE=RelWithDebInfo", "-DENABLE_GENERIC=ON"
```
Why did you enable these? Please revert that.

sonic2kk replied:
I changed these to allow the emulator to run on my phone. Otherwise when I tried to build and run it I would get errors. I'll change these back.

PEmu1 commented:
@MayImilae you want to review the visual changes?

riking commented on Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/ui/main/MainPresenter.java:
```diff
@@ -28,7 +28,7 @@ public MainPresenter(MainView view)
 	{
 		// TODO Rather than calling into native code, this should use the commented line below.
-		// String versionName = BuildConfig.VERSION_NAME;
+		//String versionName = BuildConfig.VERSION_NAME;
```
extraneous diff

riking commented on Source/Android/app/src/main/res/layout/activity_main.xml:
```diff
@@ -36,13 +41,14 @@
-        android:layout_height="wrap_content"
         android:layout_width="wrap_content"
-        android:layout_margin="32dp"
+        android:layout_height="wrap_content"
```
extraneous diff

riking commented on Source/Android/app/src/main/res/menu/menu_game_grid.xml:
```diff
@@ -10,9 +10,9 @@
         android:title="@string/grid_menu_video_settings"
-        android:icon="@drawable/ic_settings_graphics"
-        app:showAsAction="ifRoom"/>
+        app:showAsAction="ifRoom" />
```
again, space was added + the :icon was moved to a different line?

sonic2kk replied:
My apologies, but I can't see the space you're referring to?

JosJuice replied:
The space is before the `/>`

sonic2kk replied:
Oh wow, I was looking for some time and I didn't see it. Thanks! Pushing the latest edits now.

riking commented on Source/Android/app/src/main/res/layout/activity_main.xml:
```diff
@@ -14,17 +14,22 @@
+            android:paddingEnd="4dp"
+            android:paddingTop="4dp"
+            app:layout_scrollFlags="scroll|enterAlways"
+            app:popupTheme="@style/ThemeOverlay.AppCompat.Light" />
```
popoutTheme and layout_scrollFlags have no changes except getting swapped around

sonic2kk commented:
@riking most of these strange differences are a result of Android Studio, as I didn't edit the XML file except for changing the Floating Action Button to 16dp rather than 32dp. The rest must have been changed by Android Studio, and my apologies for not looking into this further.

However the first difference you mentioned was actually my fault. I accidentally uncommented that line and when I recommented it I put a space in it seems. I'll get to work on fixing these issues now.

sonic2kk commented:
Just realised I forgot to revert a change I made to the action bar height, I'll edit this back now.

degasus commented on Source/Android/app/build.gradle:
```diff
@@ -65,7 +65,7 @@ android {
             cmake {
                 arguments "-DANDROID_STL=c++_static", "-DCMAKE_BUILD_TYPE=RelWithDebInfo" // , "-DENABLE_GENERIC=ON"
-                abiFilters "arm64-v8a" //, "armeabi-v7a", "x86_64", "x86"
+                abiFilters "arm64-v8a", "armeabi-v7a" //, "x86_64", "x86"
```
IIRC we fail in cmake for 32 bit if the generic flag isn't set.

degasus commented:
The visual change looks good IMO. Is the old icon still in use by the android TV - if not, we should remove it as well.

sonic2kk commented:
I had to do a reinstall of Android Studio, and then for some reason it didn't recognise my Dolphin project on my computer. I am currently trying to get to the point where I can be up to date with the project again. I think I have gotten there now, but I am still unsure. Android Studio and the Dolphin project are not wanting to co-operate. Once I get it working I will try to check if the icon is still in use by Android TV.

sonic2kk commented:
I was unable to setup an Android TV virtual device and I do not own any Android TV devices to test on, but I changed the Android TV icon but kept the same name in the hopes that it will work. If someone can test this I would appreciate it.

JosJuice commented on Source/Android/app/build.gradle:
```diff
@@ -65,7 +65,7 @@ android {
             cmake {
                 arguments "-DANDROID_STL=c++_static", "-DCMAKE_BUILD_TYPE=RelWithDebInfo" // , "-DENABLE_GENERIC=ON"
-                abiFilters "arm64-v8a", "x86_64" //, "armeabi-v7a", "x86"
+                abiFilters "arm64-v8a", "armeabi-v7a" //,"x86_64", "x86"
```
Still wrong.

sonic2kk replied:
Whoops, I'll fix that ASAP. I have to keep changing these in order to get the project to run on my phone.

sonic2kk commented:
Hopefully now that build.gradle file is fixed. I also changed the theme and UI in the add directory screen, and I added an icon to complete the TODO in menu_add_directory.xml (https://material.io/icons/#ic_arrow_upward). The colour of the FAB was changed to more closely match the toolbar/appbar also.

Here is a picture of the new main activity screen: http://imgur.com/a/VsVKd

MayImilae commented:
With this PR, you are bundling a TON of different things together... It makes it much harder to review and harder to merge. :/ I'm not a coder, but I'll go through it and address what concerns me.

Honestly, I'm not a big fan of any of the existing android icons. They were made without my input using stock option, so um, bleh. Really I just need to make a PR and redo all of the buttons! As for the video icon this PR does... Well, it's better than what's there... but honestly they all need to be replaced. I guess consider me neutral on this. But I do plan to replace them all at some point in the near future!

As for the change to the action button... Why green? Why not make the default dolphin color match the blue that is in the top there?

JosJuice commented:
Since this PR won't get merged in its current state and the repository that it's based on has been deleted, we can't do much with this PR other than closing it. If you're still interested in getting these changes into Dolphin, feel free to split out the changes to separate PRs. I've created PR #6084 for changing the FAB placement.
